### PR TITLE
add fetch_messages function

### PR DESCRIPTION
**Pull request by shay23b**

### CODE REVIEW

File: memphis/memphis.py [hunk 1]
```diff
@@ -1040,25 +1089,80 @@ async def __consume(self, callback):
...
                 subscription_name, subscription_name
             )
+            # if consumerConstructor:
+            #     async for msg in self.consumer_dls.messages:
+            #         index_to_insert = self.dls_current_index
+            #         if index_to_insert>=10000:
+            #             index_to_insert%=10000
+            #         await callback([Message(msg, self.connection, self.consumer_group)], index_to_insert)
+                    
+            # else:
```
Comment (msiwik-epruf) [inline memphis/memphis.py:1107]: Left some commented code.

File: memphis/memphis.py [hunk 2]
```diff
@@ -996,9 +1042,12 @@ def __init__(
             error_callback = default_error_handler
         self.t_ping = asyncio.create_task(self.__ping_consumer(error_callback))
         self.start_consume_from_sequence = start_consume_from_sequence
-
         self.last_messages = last_messages
         self.context = {}
+        self.dls_messages = []
+        self.dls_current_index = 0
+        # self.dls_consumer = asyncio.create_task(self.__consume_dls(None, False))
```
Comment (msiwik-epruf) [inline memphis/memphis.py:1049]: here, too

File: memphis/memphis.py [hunk 3]
```diff
@@ -1040,25 +1089,80 @@ async def __consume(self, callback):
             else:
                 break
 
-    async def __consume_dls(self, callback):
+    async def __consume_dls(self, callback, consumerConstructor: bool):
```
Comment (msiwik-epruf) [inline memphis/memphis.py:1092]: You are not using consumerConstructor in your code.

File: memphis/memphis.py [hunk 4]
```diff
@@ -1040,25 +1089,80 @@ async def __consume(self, callback):
...
+                    )
         except Exception as e:
             print("dls", e)
             await callback([], MemphisError(str(e)))
             return
 
+    async def fetch(self):
+        """Fetch a batch of messages."""
+        messages = []
+        self.dls_consumer = asyncio.create_task(self.__consume_dls(None, False))
```
Comment (msiwik-epruf) [inline memphis/memphis.py:1129]: This false does not do anything

**Comment by shay23b** [comment]
wip

**Pull request closed without merging**